Automattic/wp-calypso
# FSE: Update Newspack sync bash script cross different platforms.

#### Changes proposed in this Pull Request

Update the way to detect the platform based on the `--version` parameter of the `sed` command. In the MacOS version, `sed` doesn't support `--version` parameter. We rely on this to detect the platform, and according to this, adapt the usage of it.

In my local env, the current way to define how to run `sed` doesn't work, producing an error:

```
> yarn run sync:newspack-blocks --branch=master
```
```
Downloading branch master
Extracting…
Syncing files to FSE…
sed: can't read s/, 'newspack-blocks' )/, 'full-site-editing' )/g: No such fi
le or directory
sed: can't read s/, 'newspack-blocks' )/, 'full-site-editing' )/g: No such fi
le or directory
sed: can't read s/, 'newspack-blocks' )/, 'full-site-editing' )/g: No such fi
le or directory
sed: can't read s/, 'newspack-blocks' )/, 'full-site-editing' )/g: No such fi
le or directory
sed: can't read s/, 'newspack-blocks' )/, 'full-site-editing' )/g: No such file or directory
```

My dev env is not running the sed from the MacOS.

```
> which sed
```

```
/usr/local/bin/sed
```

#### Testing instructions

Apply the patch and sync files. You shouldn't get any issue:

```
> yarn run sync:newspack-blocks --branch=master
```

```
yarn run v1.22.4
$ check-npm-client && ./bin/sync-newspack-blocks.sh --branch=master
Downloading branch master
Extracting…
Syncing files to FSE…
Sync done.
✨  Done in 11.05s.
retrofox@MacBook-Pro full-site-editing (update/fse-newspack-sync-bash-script-platform) 
```
